sg_jira/handlers/entity_issue_handler.py

```diff
@@ -640,7 +640,9 @@ def _sync_shotgun_cced_changes_to_jira(self, jira_issue, added, removed):
                         "Removing %s from %s watchers list."
                         % (jira_user.displayName, jira_issue)
                     )
-                    self._jira.remove_watcher(jira_issue, jira_user.user_id)
+                    # In older versions of the client (<= 3.0) we used jira_user.user_id
+                    # However, newer versions of the remove_watcher method supports name search
+                    self._jira.remove_watcher(jira_issue, jira_user.displayName)
```

Comment:
Why not use the accountId here too as bellow? Is it possible for two different Jira accounts to share the same display name? If so, this change could lead to a bug right?

Reply:
It's because `add_watcher` and `remove_watcher` implementations are different.

[add_watcher](https://github.com/pycontribs/jira/blob/3.5.0/jira/client.py#L2349) is a simple wrapper for the REST API [endpoint](https://docs.atlassian.com/software/jira/docs/api/REST/1000.810.0/#api/2/issue-addWatcher) (and it's not clear what do they want as input). 

[remove_watcher](https://github.com/pycontribs/jira/blob/3.5.0/jira/client.py#L2363) performs an internal lookup based on a query string (`displayName` in this case) to get the user id. Then it uses the REST API [endpoint](https://docs.atlassian.com/software/jira/docs/api/REST/1000.810.0/#api/2/issue-removeWatcher) depending if we're on a cloud or server environment.

According to the [internal implementation](https://github.com/pycontribs/jira/blob/3.5.0/jira/client.py#L1881), if two display names exist, then the first one will be returned.

Reply:
I don't like the displayName approach because of the reason given above. I can live with it if we document this limitation and create a ticket about it.

Reply:
Remove watcher by displayName sounds designed by JIRA api. I don't think we can fix it.
https://github.com/pycontribs/jira/blob/da00ec08a69d002ebb6e6292322ef527632fce42/jira/client.py#L2355C37-L2355C37
https://docs.atlassian.com/software/jira/docs/api/REST/8.13.6/#api/2/issue-removeWatcher

Reply:
@julien-lang created SG-31742 to work on the docs update.